### PR TITLE
Fix %AddJar parsing of JAR URL.  Don't attempt classpath reload if jar file not .jar or .zip.

### PR DESCRIPTION
**Pull request by jtyberg**
Fixes issue #181

### CODE REVIEW

File: scala-interpreter/src/main/scala/com/ibm/spark/kernel/interpreter/scala/ScalaInterpreter.scala [hunk 1]
```diff
@@ -154,6 +154,7 @@ class ScalaInterpreter(
     val platform = sparkIMain.global.platform.asInstanceOf[JavaPlatform]
 
     val newClassPath = mergeJarsIntoClassPath(platform, jars:_*)
+    logger.info(s"newClassPath: ${newClassPath}")
```
Comment (lbustelo) [inline scala-interpreter/src/main/scala/com/ibm/spark/kernel/interpreter/scala/ScalaInterpreter.scala]: Can you put this log as a debug or trace?

Comment (jtyberg) [inline scala-interpreter/src/main/scala/com/ibm/spark/kernel/interpreter/scala/ScalaInterpreter.scala]: Yeah, I saw that after I pushed.  I changed to debug and removed the log statements below.

File: kernel/src/main/scala/com/ibm/spark/magic/builtin/AddJar.scala [hunk 2]
```diff
@@ -85,7 +98,12 @@ class AddJar
     }
 
     // Get the destination of the jar
-    val HtmlJarRegex(jarName) = jarRemoteLocation
+    val jarName = getFileFromLocation(jarRemoteLocation)
+
+    // Ensure the URL actually contains a jar or zip file
+    if (!jarName.endsWith(".jar") && !jarName.endsWith(".zip")) {
+        throw new IllegalArgumentException(s"The jar file $jarName must end in .jar or .zip.")
```
Comment (lbustelo) [inline kernel/src/main/scala/com/ibm/spark/magic/builtin/AddJar.scala]: Did you test that this exception is caught somewhere and does not end up killing the kernel?

Comment (jtyberg) [inline kernel/src/main/scala/com/ibm/spark/magic/builtin/AddJar.scala]: Yup.  It ends up propagating back to the notebook user, as I might expect.

addjar_ziporjar

**Comment by Lull3rSkat3r** [comment]
LGTM!

**Pull request merged**